### PR TITLE
Print additional info for Plan 1 rechits

### PR DESCRIPTION
**Pull request by igv4321**
Add the code to print the depth of initial rechits from which the plan 1 rechit has been made.

This PR will not modify any relval results.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: DataFormats/HcalRecHit/test/HcalRecHitDump.cc [hunk 1]
```diff
@@ -55,13 +55,26 @@ namespace {
...
+        if (plan1 && i.isMerged())
+        {
+            // This is a "Plan 1" combined rechit
+            std::vector<HcalDetId> ids;
+            i.getMergedIds(&ids);
+            const unsigned n = ids.size();
+            s << "; merged " << n << ": ";
+            for (unsigned j=0; j<n; ++j)
+            {
+                if (j) s << ", ";
```
Comment (perrotta) [inline DataFormats/HcalRecHit/test/HcalRecHitDump.cc]: implicit conversions should be better avoided 
if (j!=0) ...
But ok, this is test code, I won't be strict on it...

**Comment by perrotta** [comment]
+1
Just prints out additional infos in a test module.
No effect at all on reco objects

**Comment by davidlange6** [comment]
+1

**Pull request merged**